test/src/hidden_ranges/hidden_ranges_selection_test.dart

```diff
@@ -0,0 +1,211 @@
+// ignore_for_file: avoid_redundant_argument_values
+// ignore_for_file: prefer_const_constructors
+
+import 'package:flutter/widgets.dart';
+import 'package:flutter_code_editor/src/code_field/text_selection.dart';
+import 'package:flutter_code_editor/src/hidden_ranges/hidden_ranges.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+import 'common.dart';
+
+const affinities = [TextAffinity.upstream, TextAffinity.downstream];
+
+void main() {
+  group('HiddenRanges. cutSelection, recoverSelection', () {
+    test('No ranges - No changes', () {
+      const selections = [
+        TextSelection.collapsed(offset: -1),
+        TextSelection.collapsed(offset: 0),
+        TextSelection(
+          baseOffset: 5,
+          extentOffset: 10,
+          affinity: TextAffinity.downstream,
+          isDirectional: true,
+        ),
+        TextSelection(
+          baseOffset: 5,
+          extentOffset: 10,
+          affinity: TextAffinity.upstream,
+          isDirectional: true,
+        ),
+        TextSelection(
+          baseOffset: 5,
+          extentOffset: 10,
+          affinity: TextAffinity.downstream,
+          isDirectional: false,
+        ),
+        TextSelection(
+          baseOffset: 5,
+          extentOffset: 10,
+          affinity: TextAffinity.upstream,
+          isDirectional: false,
+        ),
+      ];
+
+      for (final selection in selections) {
+        expect(
+          HiddenRanges.empty.cutSelection(selection),
+          selection,
+        );
+        expect(
+          HiddenRanges.empty.cutSelection(selection.reversed),
+          selection.reversed,
+        );
+
+        expect(
+          HiddenRanges.empty.recoverSelection(selection),
+          selection,
+        );
+        expect(
+          HiddenRanges.empty.recoverSelection(selection.reversed),
+          selection.reversed,
+        );
+      }
+    });
+
+    group('Collapsed', () {
+      test('before ranges', () {
```

Comment:
This 4 tests almost the same. Suggest to extract to a function and pass numbers as parameters

Reply:
Merged with the test below.